### PR TITLE
dcp-810 Make platform versions consistent

### PR DESCRIPTION
**Pull request by MightyAx**
GitHub: ebi-ait/dcp-ingest-central#810
ZenHub: dcp-810

- [x] Upgrade to Angular 12.1.5
- [x] Upgrade to node 14.15.5
- [x] introduce common .nvmrc file for easy synchronisation of node version across developers
- [x] Use Common Node/Angular versions in gitlab tests.
- [x] Use locked dependancies
- ~~Cache the dependencies for quicker CI builds~~
    - The GitLab local cache is garbage, only use if a remote cache is available.
- [x] Report Test Success to GitLab

Screenshot 2022-07-28 at 12 47 13

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -2,7 +2,17 @@ include:
   remote: 'https://raw.githubusercontent.com/ebi-ait/gitlab-ci-templates/master/build-release-deploy.yml'
 
 Test:
-  image: quay.io/ebi-ait/ingest-base-images:trion_ng-cli-karma_11.2.6
+  image: quay.io/ebi-ait/ingest-base-images:node_14.15.5-alpine
```
Comment (amnonkhen) [inline .gitlab-ci.yml:5]: Can we use an environment variable here that would take the value from the `package.json` or `.nvmrc` file or at least have the gitlab job fail if there is a version mismatch? I think yarn/npm will fail if the node engine does not match the *engine* configuration in `package.json`.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -2,7 +2,17 @@ include:
...
+  stage: test
+  before_script:
+    - yarn install --frozen-lockfile --no-progress
+    - apk add chromium
+    - export CHROME_BIN=/usr/bin/chromium-browser
   script:
-    - yarn install
-    - ng lint && ng test --watch false
+    - yarn lint
```
Comment (amnonkhen) [inline .gitlab-ci.yml:12]: Is the bulid job configured to fail when there are (too many) linting problems?
If not, why are we running it during build?

File: README.md [hunk 3]
```diff
@@ -7,26 +7,49 @@ This is the UI app for monitoring and tracking submissions to the DCP.
...
+## Further help
+
+To get more help on the Angular CLI use `ng help` or go check out the [Angular CLI README](https://github.com/angular/angular-cli/blob/master/README.md).
+
+## About Versions and Support
+The versions of **Angular**, **Node**, **TypeScript** and **RxJS** are tightly coupled with each-other, as described by [this table](https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3).
+
+Furthermore, Angular 12 is only in Long-Term Support until 2022-11-12, [source](https://angular.io/guide/releases#support-policy-and-schedule) and Node 14 is only Maintained until 2023-04-30 [source](https://github.com/nodejs/Release#nodejs-release-working-group). This project should expect to be migrated to supported versions of Node and Angular as appropriate.
+
+### Currently working versions
```
Comment (amnonkhen) [inline README.md:46]: Not sure this is needed.
This is documented in the `package.json`

File: README.md [hunk 4]
```diff
@@ -38,49 +61,28 @@ $ ng --version
...
+@angular-devkit/architect       0.1201.4
+@angular-devkit/build-angular   12.1.4
+@angular-devkit/core            12.1.4
+@angular-devkit/schematics      12.1.4
+@angular/cdk                    12.1.4
+@angular/cli                    12.1.4
+@angular/flex-layout            12.0.0-beta.35
+@angular/material               12.1.4
+@schematics/angular             12.1.4
+rxjs                            6.6.7
```
Comment (amnonkhen) [inline README.md:85]: Not sure this is necessary.

File: package.json [hunk 5]
```diff
@@ -76,16 +76,18 @@
...
     "tar": "^6.1.1",
     "ts-node": "~10.1.0",
-    "typescript": "4.3.5",
+    "typescript": ">= 4.2.4 <= 4.3.5",
     "url-parse": ">=1.5.3"
   },
   "snyk": true,
   "engines": {
     "npm": "please-use-yarn",
+    "node": ">=14.15.5 <15",
```
Comment (amnonkhen) [inline package.json:90]: I believe this should be the single source of truth for the node version, and  it should feed other locations, such as gitlab-ci.
What do you think @MightyAx?

File: .gitlab-ci.yml (hunk 1 above)
Comment (MightyAx) [reply .gitlab-ci.yml:5]: I agree that a single source of truth would be better but I can't think of a way we could do that.

Also due to our use of quay to store base images (as well as other factors) changing the angular/node version is a big deal that we shouldn't need to do that often, so I think it's okay that we have to change the config in a few places

File: .gitlab-ci.yml (hunk 2 above)
Comment (MightyAx) [reply .gitlab-ci.yml:12]: I don't know why we lint, I was just replicating the previous linting step.

File: README.md (hunk 3 above)
Comment (MightyAx) [reply README.md:46]: The dates at which the versions we are currently using leave support is not documented anywhere else.

File: package.json (hunk 5 above)
Comment (MightyAx) [reply package.json:90]: I agree that a single source of truth would be better but I can't think of a way we could do that.

**Review by ke4 — APPROVED** [review]
LGTM 👍

**Pull request merged**